### PR TITLE
Refactor CI/CD github actions

### PR DESCRIPTION
**Pull request by ikalnytskyi**
There are couple of things this patch targets to achieve:

* It strips version out from dist bundle artifact name because we want
  to easily retrieve our artifacts by tags from GitHub.

* It moves away from deprecated and archived github actions to create
  releases and upload assets towards third party one which is
  maintained and recommended.

* CD is now triggered on every push on master to build and prerelease the
  nightly version.

* Created releases are created with changelogs now.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            VERSION="${GITHUB_REF#refs/tags/}"
+          fi
+
+          echo ::set-output name=version::${VERSION}
+          echo ::set-output name=prerelease::${PRERELEASE}
+        id: metadata
+
+      # Since 'softprops/action-gh-release@v1' does not support release
+      # re-creation, we have to remove 'nightly' release first.
+      - name: Remove nightly release
```
Comment (ikalnytskyi) [inline .github/workflows/cd.yml:43]: I feel like I'm going to write my own GitHub action for the whole "create release" thing :D

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Create xsnippet-web.tar.gz
+        run: tar cvzf xsnippet-web.tar.gz -C dist/ .
+
+      - name: Setup release metadata
+        run: |
+          PRERELEASE=true
+          VERSION=nightly
+
+          if [[ $GITHUB_REF =~ ^refs/tags/.* ]]; then
+            if [[ $GITHUB_REF =~ ^refs/tags/\d+.\d+.\d+$ ]]; then
```
Comment (malor) [inline .github/workflows/cd.yml]: Shouldn't this be `^refs/tags/\d+\.\d+\.\d+$`? I think `.` matches any single character, while `\.` matches the dot character specifically?

**Review by malor — APPROVED** [review]
Thank you for cleaning this up! :+1:

File: .github/workflows/cd.yml (hunk 2 above)
Comment (ikalnytskyi) [reply .github/workflows/cd.yml]: Thanks! What's worse is that `\d` does not work in bash. :thinking:  I tested this script manually and replaced `\d` with `[0-9]` groups. I also adjusted a bit the regular expression to take into account `1.0` versions (i.e. without patch version).

**Pull request merged**